test/jsonTester.cpp

```diff
@@ -0,0 +1,48 @@
+#include "../Json.h"
+
+#include <gtest/gtest.h>
+
+TEST(JsonTest, ParseString)
+{
+    std::map<std::string, std::string> data = Json::ParseString("{   \"hp\" : 200  ,   \"name\" : \"Kakarott\"  ,   \"attackcooldown\": 1.0  ,   \"dmg\" : 90   }");
+    ASSERT_EQ(data["name"], "Kakarott");
+    ASSERT_EQ(data["hp"], "200");
+    ASSERT_EQ(data["dmg"], "90");
+    ASSERT_EQ(data["attackcooldown"], "1.0");
+}
+TEST(JsonTest, doesnotexistFile)
+{
+    ASSERT_THROW(Json::ParseFile("../unit/doesnotexist.json"), std::runtime_error);
+}
+TEST(JsonTest, existFile)
+{
+    ASSERT_NO_THROW(Json::ParseFile("../unit/unit1.json"));
+}
+TEST(JsonTest, ValidatorMissingChars)
+{
+    ASSERT_THROW(Json::ParseString("    \"name\" : \"Kakarott\",    \"hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0 }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    name\" : \"Kakarott\",    \"hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name : \"Kakarott\",    \"hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\"  \"Kakarott\",    \"hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : Kakarott\",    \"hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : \"Kakarott,    \"hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : \"Kakarott\"    \"hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : \"Kakarott\",    hp\" : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : \"Kakarott\",    \"hp : 200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : \"Kakarott\",    \"hp\"  200,    \"dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : \"Kakarott\",    \"hp\" : 200    \"dmg\" : 90, \"attackcooldown\": 1.0  "), std::runtime_error);
+    ASSERT_THROW(Json::ParseString("{    \"name\" : \"Kakarott\",    \"hp\" : 200    dmg\" : 90, \"attackcooldown\": 1.0  }"), std::runtime_error);
```

Comment:
Adhatnál még ide egy két olyan teszt esetet ahol, nem csak egy szükséges karakter hiányzik, hanem több is.
Pl: hiányzik két idéző jel, vagy kettőspont és és kapcsos zárójel.